packages/tasks/src/tasks/text-generation/inference.ts

```diff
@@ -5,204 +5,94 @@
  */
 
 /**
- * Inputs for Text Generation inference
+ * Text Generation Input
  */
 export interface TextGenerationInput {
-	/**
-	 * The text to initialize generation with
-	 */
 	inputs: string;
-	/**
-	 * Additional inference parameters
-	 */
-	parameters?: TextGenerationParameters;
-	/**
-	 * Whether to stream output tokens
-	 */
-	stream?: boolean;
+	parameters?: TextGenerationInputGenerateParameters;
 	[property: string]: unknown;
 }
 
-/**
- * Additional inference parameters
- *
- * Additional inference parameters for Text Generation
- */
-export interface TextGenerationParameters {
-	/**
-	 * The number of sampling queries to run. Only the best one (in terms of total logprob) will
-	 * be returned.
-	 */
+export interface TextGenerationInputGenerateParameters {
 	best_of?: number;
-	/**
-	 * Whether or not to output decoder input details
-	 */
```

Comment:
How would we proceed to generate JSDocs like this?
I think this is very useful for downstream consumers

Reply:
Documentation should be auto-generated from the TGI's specs. Currently `GenerateParameters` is not properly documented there which creates those deletions. I opened a PR in TGI to address this (https://github.com/huggingface/text-generation-inference/pull/1798). Once it's merged, we should be able to re-generate those without additional effort.

It is already the case for `chat_completion` parameters.